client/sites-dashboard/components/wpcom-smp-commands.tsx

```diff
@@ -426,6 +426,76 @@ export const useCommandsArrayWpcom = ( {
 			},
 			icon: addNewSiteIcon,
 		},
+		{
+			name: 'manageSettingsWriting',
+			label: __( 'Manage writing settings' ),
+			searchLabel: _x( 'Manage blog writing settings', 'search label for command palette' ),
+			context: 'Managing settings for writing the blog',
+			callback: setStateCallback( 'manageSettingsWriting' ),
```

Comment:
What does this callback do and is it needed?

Reply:
The callback is executed when this command is selected by clicking on it or pressing enter when highlighted.

The function `setStateCallback` is our way to set the "selectedCommandName" and present the sites selector as a nested step.
The `siteFunctions.onClick` is executed when selecting one of those nested commands-sites, and the `selectedCommandName` is used to find the "parent" command that was selected.

If you implement other command that is just navigational and do not need to pick a site, the you'll use only the callback.
The callback allows us the freedom to execute any code, like opening modals, or making API calls.